### ISSUE TITLE
[4.x]: Cannot assign null to property exception when trying to save Sale/Promotion on fresh install

### ISSUE THREAD

**Issue by bliswebagency** [craftcms/commerce#3364]
### Description

After a fresh Craft and Commerce 4 install without any other plugins, trying to save a Sale results in cannot assign null to property error

### Steps to reproduce

1. Set up craft and commerce store. Add test product
2. Create a new Sale setting values: title, description, matching item, effect (eg. reduce by 50%), Enable switched on

### Expected behavior

No exception and Sale is saved

### Actual behavior

Exception occurs and Sale is not saved

```
TypeError: Cannot assign null to property craft\commerce\models\Sale::$categoryRelationshipType of type string in /apps/commerce-craft/craft/vendor/craftcms/commerce/src/controllers/SalesController.php:145
Stack trace:
#0 [internal function]: craft\commerce\controllers\SalesController->actionSave()
#1 /apps/commerce-craft/craft/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#2 /apps/commerce-craft/craft/vendor/yiisoft/yii2/base/Controller.php(178): yii\base\InlineAction->runWithParams(Array)
#3 /apps/commerce-craft/craft/vendor/yiisoft/yii2/base/Module.php(552): yii\base\Controller->runAction('save', Array)
#4 /apps/commerce-craft/craft/vendor/craftcms/cms/src/web/Application.php(305): yii\base\Module->runAction('commerce/sales/...', Array)
#5 /apps/commerce-craft/craft/vendor/craftcms/cms/src/web/Application.php(606): craft\web\Application->runAction('commerce/sales/...', Array)
#6 /apps/commerce-craft/craft/vendor/craftcms/cms/src/web/Application.php(284): craft\web\Application->_processActionRequest(Object(craft\web\Request))
#7 /apps/commerce-craft/craft/vendor/yiisoft/yii2/base/Application.php(384): craft\web\Application->handleRequest(Object(craft\web\Request))
#8 /apps/commerce-craft/craft/web/index.php(12): yii\base\Application->run()
#9 {main}
```

### Craft CMS version

4.6.0

### Craft Commerce version

4.3.3

### PHP version

8.1.27

### Operating system and version

Linux 4.4.0-210-generic

### Database type and version

MySQL 5.7.33

### Image driver and version

 	GD 8.1.27

### Installed plugins and versions

-
Commerce 4.3.3 and nothing else

**Comment by lukeholder** [comment]
Could you try clearing all template caches and let me know if the error persists? 
It sounds like your sales control panel page is cached and is missing the new input option under the "Only match purchasables related to…" option in the "Matching Items" tab.

**Comment by lukeholder** [comment]
Thanks for reporting this. It also affects discounts, and we will fix them both shortly and let you know.

**Comment by lukeholder** [comment]
This is fixed in Commerce 4.4.1.1

Thanks!
